### ISSUE TITLE
Save a checkpoint based on best validation error

### ISSUE THREAD

**Issue by ybzhou** [google/seq2seq#196]
It seems that there is no easy way to save model to a checkpoint that result in the best validation error apart from saving all checkpoints.

**Comment by zhyack** [comment]
Maybe you can try decreasing your learning rate when get a rise in validation error, which help your model stop near the best checkpoint. However, it need you to run the validation set more frequently, and to tune a  rate for decreasing learning rate.

**Comment by ybzhou** [comment]
My point is that, I have to manually look through the logs to identify the optimal checkpoint. But I think it can be much simpler by just saving an additional "best_validation_error_checkpoint" file.
